utils.js

```diff
@@ -41,6 +42,83 @@ function calculateDimensions({
   };
 }
 
+function getStyleDimensions(
+  style,
+  horizontal = false,
+) {
+  let space1 = 0;
+  let space2 = 0;
+  let maxStyleDimension = undefined;
+  if (style) {
+    const flatStyle = Array.isArray(style) ? StyleSheet.flatten(style) : style;
+    let sMaxDimensionXY = 'maxWidth';
+    let sPaddingXY = 'paddingHorizontal';
+    let sPadding1 = 'paddingLeft';
+    let sPadding2 = 'paddingRight';
+    if (horizontal) {
+      sMaxDimensionXY = 'maxHeight';
+      sPaddingXY = 'paddingVertical';
+      sPadding1 = 'paddingTop';
+      sPadding2 = 'paddingBottom';
+    }
+
+    if (flatStyle[sMaxDimensionXY] && typeof flatStyle[sMaxDimensionXY] === 'number') {
+      maxStyleDimension = flatStyle[sMaxDimensionXY];
+    }
+
+    const padding = flatStyle[sPaddingXY] || flatStyle.padding;
+    const padding1 = flatStyle[sPadding1] || padding || 0;
+    const padding2 = flatStyle[sPadding2] || padding || 0;
+    space1 = (typeof padding1 === 'number' ? padding1 : 0);
+    space2 = (typeof padding2 === 'number' ? padding2 : 0);
+  }
+  return { space1, space2, maxStyleDimension };
+}
+
+function adjustDimension({
```

Comment:
I would call this `getStyleAdjustedTotalDimensions`, and the first parameter can `totalDimension` instead of `newTotalDimension`